src/library_fsfs.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * @license
+ * Copyright 2022 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+mergeInto(LibraryManager.library, {
+  $FSFS__deps: ['$FS', '$MEMFS', '$PATH'],
+  $FSFS__postset: function() {
+    return '';
+  },
+  $FSFS: {
+    DIR_MODE: Number("{{{ cDefine('S_IFDIR') }}}") | 511 /* 0777 */,
+    FILE_MODE: Number("{{{ cDefine('S_IFREG') }}}") | 511 /* 0777 */,
```

Comment:
This looks a little funny, but it's wrapped in a string because otherwise my editor barfs on parsing the rest of the file, and it was hard to work like that.